sdk/include/platform/sunburst/platform-simulation_exit.hh

```diff
@@ -0,0 +1,25 @@
+// Copyright Microsoft and CHERIoT Contributors.
+// SPDX-License-Identifier: MIT
+
+#pragma once
+
+#ifdef SIMULATION
+#	include <stdint.h>
+#	include <platform-uart.hh>
+#	include <string_view>
+
+static void platform_simulation_exit(uint32_t code)
+{
+#	if DEVICE_EXISTS(uart0)
+	auto uart = MMIO_CAPABILITY(Uart, uart0);
+#	elif DEVICE_EXISTS(uart)
+	auto uart = MMIO_CAPABILITY(Uart, uart);
+#	endif
+	// The following magic string will cause sonata simulator to exit.
+	using namespace std::string_view_literals;
```

Comment:
I am not convinced that this is more readable (it’s definitely longer) than just using the long-form constructor.

Reply:
If I do `std::string_view s {"magic string"}` there is a call to `strlen` which means I have to link the string library into hello_world and any other example that doesn't already have it. Using a string_view literal (or just writing the C-style loop) avoids this.

Reply:
Computers :-(